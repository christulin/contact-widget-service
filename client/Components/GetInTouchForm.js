import React from 'react';

const GetInTouchForm = () => {


  return (
    <h1>GetInTouchForm here</h1>
  )
}

export default GetInTouchForm;