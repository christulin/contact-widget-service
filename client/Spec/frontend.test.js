import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import React from 'react';
import renderer from 'react-test-renderer';
import GetInTouchForm from '../Components/GetInTouchForm';
import { shallow } from 'enzyme';
import sinon from 'sinon';

Enzyme.configure({ adapter: new Adapter() });

describe('Get In Touch Form', () => {


  test('Form loads correctly', () => {
    const component = renderer.create(
      <GetInTouchForm />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

  });

  test('form fields work properly', () => {

    const wrapper = shallow(<GetInTouchForm />);

    const items = wrapper.find('div.form-item')
    expect(items.length).toEqual(5);

  });

})
