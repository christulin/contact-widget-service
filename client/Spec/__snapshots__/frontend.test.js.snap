// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form loads correctly 1`] = `
<div
  className="get-in-touch-form-wrapper"
>
  <button
    className="close-form"
    onClick={[Function]}
  >
    X
  </button>
  <h3>
    Get in touch
  </h3>
  <form>
    <p>
      Fill out your info below and we'll get you connected with one of our representatives.
    </p>
    <div
      className="form-item"
    >
      <div
        className="field"
      >
        <label
          htmlFor="fullName"
        >
          Full name
        </label>
        <input
          id="fullName"
          name="fullName"
          onBlur={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
        />
        <div
          className="error-message"
          hidden={true}
        >
          First and last, please.
        </div>
      </div>
    </div>
    <div
      className="form-item"
    >
      <div
        className="field"
      >
        <label
          htmlFor="email"
        >
          Work Email
        </label>
        <input
          id="email"
          name="email"
          onBlur={[Function]}
          onClick={[Function]}
          required={true}
          type="email"
        />
        <div
          className="error-message"
          hidden={true}
        >
          We need this for confirmations.
        </div>
      </div>
    </div>
    <div
      className="form-item"
    >
      <div
        className="field"
      >
        <label
          htmlFor="phone"
        >
          Phone Number
        </label>
        <input
          id="phone"
          name="phone"
          onBlur={[Function]}
          onClick={[Function]}
          required={true}
          type="phone"
        />
        <div
          className="error-message"
          hidden={true}
        >
          Please give us a way to get in contact.
        </div>
      </div>
    </div>
    <div
      className="cell-container"
    >
      <div
        className="form-item cell"
      >
        <div
          className="cell-wrap"
        >
          <label
            htmlFor="moveInDateBtn"
          >
            Move-in date
          </label>
          <button
            className="move-in"
            id="moveInDataBtn"
            type="button"
          >
            Jan 21
          </button>
          <input
            hidden={true}
            id="moveInDate"
            name="moveInDate"
            type="text"
          />
        </div>
      </div>
      <div
        className="form-item cell"
      >
        <div
          className="cell-wrap"
        >
          <label
            htmlFor="desiredCapacity"
          >
            Number of People
          </label>
          <input
            id="desiredCapacity"
            required={true}
            type="number"
            value="1"
          />
          <button>
            -
          </button>
          <button>
            +
          </button>
        </div>
      </div>
    </div>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`Get In Touch Form Form loads correctly 1`] = `
<div
  className="get-in-touch-form-wrapper"
>
  <button
    className="close-form"
    onClick={[Function]}
  >
    X
  </button>
  <h3>
    Get in touch
  </h3>
  <form>
    <p>
      Fill out your info below and we'll get you connected with one of our representatives.
    </p>
    <div
      className="form-item"
    >
      <div
        className="field"
      >
        <label
          htmlFor="fullName"
        >
          Full name
        </label>
        <input
          id="fullName"
          name="fullName"
          onBlur={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
        />
        <div
          className="error-message"
          hidden={true}
        >
          First and last, please.
        </div>
      </div>
    </div>
    <div
      className="form-item"
    >
      <div
        className="field"
      >
        <label
          htmlFor="email"
        >
          Work Email
        </label>
        <input
          id="email"
          name="email"
          onBlur={[Function]}
          onClick={[Function]}
          required={true}
          type="email"
        />
        <div
          className="error-message"
          hidden={true}
        >
          We need this for confirmations.
        </div>
      </div>
    </div>
    <div
      className="form-item"
    >
      <div
        className="field"
      >
        <label
          htmlFor="phone"
        >
          Phone Number
        </label>
        <input
          id="phone"
          name="phone"
          onBlur={[Function]}
          onClick={[Function]}
          required={true}
          type="phone"
        />
        <div
          className="error-message"
          hidden={true}
        >
          Please give us a way to get in contact.
        </div>
      </div>
    </div>
    <div
      className="cell-container"
    >
      <div
        className="form-item cell"
      >
        <div
          className="cell-wrap"
        >
          <label
            htmlFor="moveInDateBtn"
          >
            Move-in date
          </label>
          <button
            className="move-in"
            id="moveInDataBtn"
            type="button"
          >
            Jan 21
          </button>
          <input
            hidden={true}
            id="moveInDate"
            name="moveInDate"
            type="text"
          />
        </div>
      </div>
      <div
        className="form-item cell"
      >
        <div
          className="cell-wrap"
        >
          <label
            htmlFor="desiredCapacity"
          >
            Number of People
          </label>
          <input
            id="desiredCapacity"
            required={true}
            type="number"
            value="1"
          />
          <button>
            -
          </button>
          <button>
            +
          </button>
        </div>
      </div>
    </div>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
